GateAdventures/Scenes/HelloScene.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using GateAdventures.Engine.Scenes;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Microsoft.Xna.Framework.Input;
+
+namespace GateAdventures;
+
+// Example scene
+public class HelloScene : Scene
+{
+	private Vector2 _ballVelocity;
+	private Vector2 _ballPosition;
+	private Texture2D _ballTexture;
+	private MouseState _previousMouseState;
+
+	public override void Initialize(GameServiceContainer services)
+	{
+		base.Initialize(services);
+
+		_ballVelocity = Vector2.Zero;
+		_ballPosition = GraphicsDevice.Viewport.Bounds.Center.ToVector2();
+		_previousMouseState = Mouse.GetState();
+	}
+
+	public override void LoadContent()
+	{
+		_ballTexture = Content.Load<Texture2D>("RedBall");
+	}
+
+	public override void HandleInput()
+	{
+		MouseState currentMouseState = Mouse.GetState();
+		Vector2 mousePosition = new(currentMouseState.X, currentMouseState.Y);
+
+		// Calculate steering force with maximum magnitude of 4 in 1 frame
+		Vector2 desiredDirection = Vector2.Subtract(mousePosition, _ballPosition);
+		float desiredMagnitude = Math.Min(Vector2.Distance(mousePosition, _ballPosition), 4);
+		desiredDirection.Normalize();
+		desiredDirection = Vector2.Multiply(desiredDirection, desiredMagnitude);
+
+		// Apply steering force
+		_ballVelocity += desiredDirection;
+
+		// Limit velocity to a maximum magnitude of 25
+		float velocityMagnitude = Math.Min(Vector2.Distance(Vector2.Zero, _ballVelocity), 25);
+		_ballVelocity.Normalize();
+		_ballVelocity = Vector2.Multiply(_ballVelocity, velocityMagnitude);
+
+		// Apply friction
+		_ballVelocity = Vector2.Multiply(_ballVelocity, 0.9f);
+
+		if (_previousMouseState.LeftButton == ButtonState.Pressed && currentMouseState.LeftButton == ButtonState.Released)
+		{
+			SceneManager.Stop(this);
+			SceneManager.StartOverlay(new WorldScene());
+			return;
+		}
+
+		_previousMouseState = currentMouseState;
```

Comment:
This is the same as in `WorldScene`, I know it's just an example, but we should move this to a helper function. It'll be good practice for when we do have shared input functionality between scenes

Reply:
Yes all this code is nearly identical but I only intend these classes to be short lived examples until we get farther along! I am not certain a helper function would be worth it right now

Reply:
Works for me